### PR TITLE
add sample for stepfunctions using lambdas and localstack

### PR DESCRIPTION
**Pull request by HarshCasper**

### CODE REVIEW

File: stepfunctions-lambda/Makefile [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	rm package.zip lambda_function.py
+
+	cp -f lambda_combine.py lambda_function.py
+	zip package.zip lambda_function.py
+
+	awslocal lambda create-function \
+  		--function-name combine \
+  		--role arn:aws:iam::000000000000:role/lambda-demo \
+  		--runtime python3.8 --timeout 10 --memory-size 128 \
```
Comment (dominikschubert) [inline stepfunctions-lambda/Makefile:45]:
```suggestion
  		--runtime python3.9 --timeout 10 --memory-size 128 \
```

Comment (dominikschubert) [reply stepfunctions-lambda/Makefile:45]: same for the others. Better to use the newest right now

File: stepfunctions-lambda/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Start the execution:
+
+```sh
+awslocal stepfunctions start-execution \
+  --state-machine-arn arn:aws:states:us-east-1:000000000000:stateMachine:step-demo \
+  --input '{"adam": "LocalStack", "cole": "Stack"}'
+```
+
+This creates and invokes the flow between the three Lambda functions we created using LocalStack earlier.
```
Comment (dominikschubert) [inline stepfunctions-lambda/README.md:43]: Feels like this is missing some context about what this sample actually does and why. It's really hard to see the intent here. E.g. why is `adam` set to `LocalStack` and `cole` to `Stack`?

File: stepfunctions-lambda/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+make create-lambdas
+```
+
+## Setting up and running Step Function
+
+Create the Step Function:
+
+```sh
+awslocal stepfunctions create-state-machine --name step-demo \
+  --definition "$(cat step-definition.json)" \
```
Comment (dominikschubert) [inline stepfunctions-lambda/README.md:31]: Why is this not using the same pattern as above in the Makefile? (`file://....`)

File: stepfunctions-lambda/lambda_cole.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+def lambda_handler(event, context):
+    print(event)
+    return event["input"]["lewis"]
```
Comment (dominikschubert) [inline stepfunctions-lambda/lambda_cole.py:3]: should this be cole instead of lewis?

**Review by dominikschubert — APPROVED** [review]
Added some comments. I feel like this sample is generally missing a proper scenario/intent. It's not very clear what kind of process this tries to model and why that would be particularly useful for a user.

Since it's just a sample this wouldn't block a merge though :+1:

File: stepfunctions-lambda/Makefile [hunk 5]
```diff
@@ -0,0 +1,69 @@
+export AWS_ACCESS_KEY_ID ?= test
+export AWS_SECRET_ACCESS_KEY ?= test
+export AWS_DEFAULT_REGION = us-east-1
```
Comment (joe4dev) [inline stepfunctions-lambda/Makefile:3]: Do we intentionally overwrite the default region here or should it be `?=` ?

Comment (HarshCasper) [reply stepfunctions-lambda/Makefile:3]: On standard template, its `export AWS_DEFAULT_REGION=us-east-1`

File: stepfunctions-lambda/README.md (hunk 3 above)
Comment (HarshCasper) [reply stepfunctions-lambda/README.md:31]: Its not needed here

**Pull request merged**